src/python/pants/backend/javascript/nodejs_project.py

```diff
@@ -190,16 +190,24 @@ async def find_node_js_projects(
     nodejs: NodeJS,
     resolve_names: UserChosenNodeJSResolveAliases,
 ) -> AllNodeJSProjects:
-    # Note: If pnpm_workspace.yaml is present for an npm-managed project, it will override the package.json["workspaces"] setting, which is not intuitive
-    # pnpm_workspace.yaml should only be used for pnpm projects - see https://github.com/pantsbuild/pants/issues/21134
-    project_paths = (
-        (
-            ProjectPaths(pkg.root_dir, ["", *pkg.workspaces])
-            if pkg not in pnpm_workspaces
-            else ProjectPaths(pkg.root_dir, ["", *pnpm_workspaces[pkg].packages])
-        )
-        for pkg in package_workspaces
-    )
+    project_paths = []
+    for pkg in package_workspaces:
+        if (
+            nodejs.default_package_manager
+            and PackageManager.from_string(nodejs.default_package_manager).name == "pnpm"
+            or (
+                pkg.package_manager
+                and PackageManager.from_string(pkg.package_manager).name == "pnpm"
+            )
+        ):  # case for pnpm
```

Comment:
I wonder if this should do something like:

```suggestion
        package_manager = pkg.package_manager or nodes.default_package_manager
        if package_manager and PackageManager.from_string(package_manager).name == "pnpm:
```

In particular, as written, if someone has configured `[nodejs].default_package_manager = "pnpm"` but then a particular package overrides it with `packageManager = "yarn"`, then this pnpm path would still be taken, but I think that's probably not desirable?

I also looked at similar code like the following: https://github.com/pantsbuild/pants/blob/5fd2ee62c921ed096e3f7f851db1cedc3276ad0f/src/python/pants/backend/javascript/nodejs_project.py#L111-L113

Reply:
Thanks - that's much better. I had a feeling something was off there on the weekend but couldn't see it.